syntax = "proto3";
package protos.auth;

import "google/protobuf/empty.proto";

// =============================
// üîê AuthService
// =============================
service AuthService {
  rpc Register (RegisterRequest) returns (RegisterResponse);
  rpc Login (LoginRequest) returns (LoginResponse);
  rpc VerifyToken (VerifyRequest) returns (VerifyResponse);
  rpc GetUserInfo (UserInfoRequest) returns (UserInfoResponse);
  rpc HealthCheck (google.protobuf.Empty) returns (HealthResponse);
}

// =============================
// üì© Request Messages
// =============================
message RegisterRequest {
  string email = 1;
  string password = 2;
  string name = 3;
  string username = 4;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message VerifyRequest {
  string token = 1;
}

message UserInfoRequest {
  string user_id = 1;
}

// =============================
// üì§ Response Messages
// =============================
message RegisterResponse {
  bool success = 1;
  string message = 2;
  string user_id = 3;
}

message LoginResponse {
  string access_token = 1;
  string token_type = 2;
  int64 expires_in = 3;
  string user_id = 4;
  Role role = 5;
}

message VerifyResponse {
  bool valid = 1;
  string user_id = 2;
  Role role = 3;
}

message UserInfoResponse {
  string user_id = 1;
  string email = 2;
  Role role = 3;
}

// =============================
// üî† ENUMS
// =============================
enum Role {
  FREE = 0;
  USER = 1;
  BUSINESS = 2;
  ORG = 3;
  ADMIN = 4;
}

// =============================
// ‚ù§Ô∏è HealthCheck
// =============================
message HealthResponse {
  string status = 1;
  string timestamp = 2;
}
